client/src/components/ShopContext.js

```diff
@@ -0,0 +1,175 @@
+          setCartInventory(newInventory);
+          localStorage.removeItem('cart');
+        }
+      } catch (err) {
+        console.error(err.message)
+      }
+    } else {
+      localStorage.removeItem('cart');
+    }
+  }, []);
+
+  const handleSignOut = useCallback(() => {
+    localStorage.removeItem('tokenKey');
+    localStorage.removeItem('cart');
+    setUser(undefined);
+    setCartInventory([]);
+    setOrderItems([]);
+    navigate('/')
+  }, [navigate]);
+
+  const addToCart = useCallback(async (inventoryId, quantity) => {
+    try {
+      const existingCartItem = cartInventory.find(item => item.inventoryId === inventoryId);
+      if (existingCartItem) {
+        const updatedCartItem = { ...existingCartItem, quantity: existingCartItem.quantity + quantity };
+        const updatedCartInventory = cartInventory.map(item => item.inventoryId === inventoryId ? updatedCartItem : item);
+        user && await updateCartInventory(inventoryId, updatedCartItem.quantity);
+        setCartInventory(updatedCartInventory);
+      } else {
+        const newCartItem = { inventoryId, quantity };
+        const updatedCartInventory = [...cartInventory, newCartItem];
+        user && await addCartInventory(inventoryId, quantity);
+        setCartInventory(updatedCartInventory);
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }, [cartInventory, user]);
+
+  const removeFromCart = useCallback(async (inventoryId) => {
+    const updatedCartInventory = cartInventory.filter(card => card.inventoryId !== inventoryId);
+    setCartInventory(updatedCartInventory);
+    if (user) {
+      try {
+        await deleteCartInventory(inventoryId);
+      } catch (err) {
+        console.error(err);
+      }
+    }
+  }, [cartInventory, user]);
+
+  const updateCartItemQuantity = useCallback(async (inventoryId, newQuantity) => {
+    const updatedCartInventory = cartInventory.map(card => {
+      if (card.inventoryId === inventoryId) {
+        return { ...card, quantity: newQuantity }
+      } else {
+        return card;
+      }
+    });
+    setCartInventory(updatedCartInventory);
+    try {
+      user && await updateCartInventory(inventoryId, newQuantity);
+    } catch (err) {
+      console.error(err);
+    }
+  },[cartInventory, user]);
+
+  const clearCart = useCallback(async () => {
+    setCartInventory([]);
+    try {
+      user && await clearCartInventory();
+    } catch (err) {
+      console.error(err);
+    }
+  }, [user]);
+
+  const contextValue = useMemo(() => ({
+    user,
+    handleSignIn,
+    handleSignOut,
+    cartInventory,
+    orderItems,
+    addToCart,
+    removeFromCart,
+    updateCartItemQuantity,
+    clearCart
+  }), [user, handleSignIn, handleSignOut, cartInventory, orderItems, addToCart, removeFromCart, updateCartItemQuantity, clearCart]);
+
+  if (isAuthorizing) return null;
+
+  return (
+    <ShopContext.Provider value={contextValue}>
+      {props.children}
+    </ShopContext.Provider>
+  );
+};
```

Comment:
You have a lot of values in your context. I would suggest introducing a few custom hooks to make accessing it simpler (and to hide `ShopContext`).

For example: `const user = useUser()` that just returns the user (no need to destructure); `const { handleSignIn, handleSignOut } = useAuth();` and so forth.